SimpleApp/ViewController.swift

```diff
@@ -2,34 +2,124 @@ import UIKit
+    private var password: String? = nil
+    private var isChecked = false
+
+    @IBAction func saveCredsSwitchChanged() {
+        //update UD isCheckedKey
+        //if OFF
+        // remove userNameKey & passwordKey
+        if saveCredsSwitch.isOn == false {
+            UserDefaults.standard.removeObject(forKey: Constants.userNameKey)
+            UserDefaults.standard.removeObject(forKey: Constants.passwordKey)
+        }
+    }
 
     override func viewDidLoad() {
         super.viewDidLoad()
+        usernameTextField.delegate = self
+        passwordTextField.delegate = self
+        passwordTextField.isSecureTextEntry = true
+        label.text = "SimpleApp"
+        label.textAlignment = .center
+        button.setTitle("Sign In", for: .normal)
+        savePasswordLabel.text = "Save credentials"
+        saveCredsSwitch.isOn = false
 
-        label.text = "label"
-    }
-    @IBAction func buttonPressed(_ sender: UIButton) {
-        let tabBarVC = UITabBarController()
-        let feedVC = UINavigationController(
-            rootViewController: FeedViewController(fetcher: Manager.shared)
+        usernameTextField.addTarget(
+            self,
+            action: #selector(usernameTextFieldChanged),
+            for: .editingChanged
+        )
+        passwordTextField.addTarget(
+            self,
+            action: #selector(passwordTextFieldChanged),
+            for: .editingChanged
         )
-        let galleryVC = UINavigationController(rootViewController: GalleryViewController(fetcher: SplashImageFetcher()))
 
-        feedVC.title = "Feed"
-        galleryVC.title = "Gallery"
+        restoreNamePasswordIfNeeded()
+    }
 
-        tabBarVC.setViewControllers([feedVC, galleryVC], animated: true)
+    func restoreNamePasswordIfNeeded() {
+        if let isCheckedKey = UserDefaults.standard.value(forKey: Constants.isCheckedKey) as? Bool, isCheckedKey {
+            restoreStoredNamePassword()
+        }
+    }
 
-        guard let items = tabBarVC.tabBar.items else {
-            return
+    func restoreStoredNamePassword() {
+        //set fields
+        //update switch
+        if saveCredsSwitch.isOn {
+            username = UserDefaults.standard.object(forKey: Constants.userNameKey) as? String
+            password = UserDefaults.standard.object(forKey: Constants.passwordKey) as? String
         }
+    }
 
-        let images = ["house", "star"]
+    @IBAction func buttonPressed(_ sender: UIButton) {
+        if username == Constants.username && password == Constants.password {
+            //if swiftch ON save user&pass to UD
+            if saveCredsSwitch.isOn {
+                UserDefaults.standard.set(username, forKey: Constants.userNameKey)
+                UserDefaults.standard.set(password, forKey: Constants.passwordKey)
+            }
 
-        for x in 0..<items.count {
-            items[x].image = UIImage(systemName: images[x])
+            let tabBarVC = UITabBarController()
```

Comment:
use separate method for pushing controolers

Reply:
fixed